### PR TITLE
tests: fivetran in cloudtests

### PR DESCRIPTION
**Pull request by nrainer-materialize**
Nightlies: https://buildkite.com/materialize/nightly/builds?branch=nrainer-materialize%3Atests%2Ffivetran-in-cloudtests

### CODE REVIEW

**Comment by nrainer-materialize** [comment]
@ParkMyCar, this is still a draft, but does it make sense so far?

File: misc/python/materialize/cloudtest/k8s/fivetran.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+        self.deployment = V1Deployment(
+            api_version="apps/v1",
+            kind="Deployment",
+            metadata=V1ObjectMeta(name="fivetran-destination", namespace=namespace),
+            spec=spec,
+        )
+
+
+# TODO: probably not needed
```
Comment (ParkMyCar) [inline misc/python/materialize/cloudtest/k8s/fivetran.py:93]: Correct, this is not needed! The "FivetranDestinationTester" is a testing harness provided by Fivetran and is totally separate from our testdrive integration.

File: misc/python/materialize/cloudtest/k8s/fivetran.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+
+class FivetranDestinationDeployment(K8sDeployment):
+    def __init__(self, namespace: str, apply_node_selectors: bool) -> None:
+        super().__init__(namespace)
+
+        # TODO: not working and likely not needed
+        #         volume_mounts = [
+        #             V1VolumeMount(name="data", mount_path="/data"),
+        #         ]
```
Comment (ParkMyCar) [inline misc/python/materialize/cloudtest/k8s/fivetran.py:60]: The Fivetran destination works by reading CSV files locally from disk, so testdrive needs a way to write CSVs that the destination can then read. Not totally sure how our k8s cloudtests work, but we will need a volume that can be accessed by both

**Review by ParkMyCar — COMMENTED** [review]
Woo hoo! See my one comment about the volumes, otherwise this LGTM

File: misc/python/materialize/cloudtest/k8s/fivetran.py (hunk 2 above)
Comment (nrainer-materialize) [reply misc/python/materialize/cloudtest/k8s/fivetran.py]: Will the path need to be `/share/tmp`?

**Comment by nrainer-materialize** [comment]
"Full testdrive in cloudtests" in nightly reports:
```
testdrive/fivetran-destination.td:22:1: error: connecting to fivetran destination: transport error: error trying to connect: tcp connect error: Connection refused (os error 111): tcp connect error: Connection refused (os error 111): Connection refused (os error 111)
```

@ParkMyCar: Can I find somewhere the used URL? I did not spot it in the logs. Though, I assume it is `http://fivetran-destination:6874`?

@evanharmon: Should testdrive running as standalone binary be able to access `http://fivetran-destination:6874`? `kubectl get services` contains the following port mapping:

```
nrainer@mz-nrainer ~ % kubectl get services | grep fivetran
fivetran-destination    NodePort       10.96.42.20     <none>                                   6874:32041/TCP                                                17m
```

**Pull request closed without merging**